include/arch/aarch32/smccc_macros.S

```diff
@@ -33,28 +33,28 @@
 	stm	r0!, {sp, lr}
 
 	cps	#MODE32_irq
-	mrs	r2, spsr
+	mrs	r2, spsr_fsxc
```

Comment:
hmm, OPTEE monitor seems to be doing the same thing:
https://github.com/OP-TEE/optee_os/blob/c3d0b15de7ece03f0b84b1d15379345b31674a61/core/arch/arm/sm/sm_a32.S#L26

@jenswi-linaro , @etienne-lms 
I would like some feedback from OPTEE folks regarding this.

Reply:
@soby-mathew 
Looking at Linux we see the same thing spsr is used for mrs ans spsr_fxsc is used for mrs.

```
deckard@event-horizon:~/Development/linux$ grep -r mrs * | grep spsr
arch/arm/boot/compressed/head.S:		mrs	r0, spsr
arch/arm/boot/compressed/head.S:		mrs	r0, spsr		@ Get saved CPU boot mode

deckard@event-horizon:~/Development/linux$ grep -r msr * | grep spsr
arch/arm/boot/compressed/head.S:		msr	spsr_cxsf, r9		@ Save the CPU boot mode in

```
Still seems to me as if you'd be dropping bit 8 for the mrs though...

Reply:
Ah OK I see, trying this on OPTEE we see mrs r1, spsr_fsxc is not a valid opcode.
```

core/arch/arm/kernel/thread_a32.S:290: Error: 'APSR', 'CPSR' or 'SPSR' expected -- `mrs r1,spsr_fsxc'
core/arch/arm/kernel/thread_a32.S:340: Error: 'APSR', 'CPSR' or 'SPSR' expected -- `mrs r1,spsr_fsxc'
core/arch/arm/kernel/thread_a32.S:984: Error: 'APSR', 'CPSR' or 'SPSR' expected -- `mrs r0,spsr_fsxc'
```